vars/piperLoadGlobalExtensions.groovy

```diff
@@ -0,0 +1,106 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.DebugReport
+import com.sap.piper.GenerateDocumentation
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = [
+    /** Directory where the extensions are cloned to*/
+    'globalExtensionsDirectory',
+    /** Git url of the repository containing the extensions*/
+    'globalExtensionsRepository',
+    /** Credentials required to clone the repository*/
+    'globalExtensionsRepositoryCredentialsId',
+    /** Version which should be each, e.g. the tag name*/
+    'globalExtensionsVersion'
+]
+
+@Field Set STEP_CONFIG_KEYS = []
+
+@Field Set PARAMETER_KEYS = [
+    /** This step will reinitialize the defaults. Make sure to pass the same customDefaults as to the step setupCommonPipelineEnvironment*/
+    'customDefaults',
+    /** This step will reinitialize the defaults. Make sure to pass the same customDefaultsFromFiles as to the step setupCommonPipelineEnvironment*/
+    'customDefaultsFromFiles'
+]
+
+/**
+ * This step is part of the step setupCommonPipelineEnvironment and should not be used outside independently in a custom pipeline.
+ * This step allows users to define extensions (https://sap.github.io/jenkins-library/extensibility/#1-extend-individual-stages) globally instead of in each repository.
+ * Instead of defining the extensions in the .pipeline folder the extensions are defined in another repository.
+ * You can also place a file called extension_configuration.yml in this repository.
+ * Configuration defined in this file will be treated as default values with a lower precedence then custom defaults defined in the project configuration.
+ * You can also define additional Jenkins libraries these extensions depend on using a yaml file called sharedLibraries.yml:
+ * Example:
+ * - name: my-extension-dependency
+ *   version: git-tag
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = checkScript(this, parameters)
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
```

Comment:
Perhaps take into account `STEP_CONFIG_KEYS` here, even if currently empty. (for consistency)